### PR TITLE
[RHOAIENG-1574] - Add codeflare-sdk e2e tests 

### PR DESCRIPTION
**Pull request by ChughShilpa**
[RHOAIENG-1574](https://issues.redhat.com/browse/RHOAIENG-1574)

### CODE REVIEW

**Comment by ChughShilpa** [comment]
Tests passed in jenkins build - https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/distributed-workloads/job/rhods-pipeline-dw/179/console

**Review by sutaakar — APPROVED** [review]

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot [hunk 1]
```diff
@@ -0,0 +1,54 @@
+*** Settings ***
+Documentation     Codeflare-sdk E2E tests - https://github.com/project-codeflare/codeflare-sdk/tree/main/tests/e2e
+Suite Setup       Prepare Codeflare-sdk E2E Test Suite
+Suite Teardown    Teardown Codeflare-sdk E2E Test Suite
+Library           OperatingSystem
+Library           Process
+Resource          ../../../tasks/Resources/RHODS_OLM/install/oc_install.robot
+
```
Comment (jiripetrlik) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot:8]: Add one more line.

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+*** Variables ***
+${CODEFLARE-SDK_DIR}                codeflare-sdk
+${CODEFLARE-SDK_REPO_URL}           %{CODEFLARE-SDK_REPO_URL=https://github.com/project-codeflare/codeflare-sdk.git}
+${CODEFLARE-SDK_REPO_BRANCH}        %{CODEFLARE-SDK_REPO_BRANCH=main}
+
```
Comment (jiripetrlik) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot:13]: Add one more line.

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+*** Test Cases ***
+Run TestMNISTRayClusterSDK test
+    [Documentation]    Run Go E2E test: TestMNISTRayClusterSDK
+    [Tags]  ODS-2544
+    ...     Sanity
+    ...     DistributedWorkloads
+    ...     Codeflare-sdk
+    Run Codeflare-sdk E2E Test    TestMNISTRayClusterSDK
+
```
Comment (jiripetrlik) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot:22]: Add one more line.

**Review by jiripetrlik — APPROVED** [review]

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot (hunk 3 above)
Comment (ChughShilpa) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot:24]: Added

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+*** Keywords ***
+Prepare Codeflare-sdk E2E Test Suite
+    [Documentation]    Prepare codeflare-sdk E2E Test Suite
```
Comment (jgarciao) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot:28]: All Tests Suites should call `RHOSi Setup` in the Suite Setup (example in ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/130__rhods_operator.robot)

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    ...    shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+    IF    ${result.rc} != 0
+        FAIL    Unable to clone codeflare-sdk repo ${CODEFLARE-SDK_REPO_URL}:${CODEFLARE-SDK_REPO_BRANCH}:${CODEFLARE-SDK_DIR}
+    END
+    Enable Component    codeflare
+    Enable Component    ray
+    Create Directory    %{WORKSPACE}/codeflare-sdk-e2e-logs
+
+Teardown Codeflare-sdk E2E Test Suite
```
Comment (jgarciao) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot:39]: All Tests Suites should call `RHOSi Teardown` in the Suite Setup (example in ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/130__rhods_operator.robot)

**Review by jgarciao — CHANGES_REQUESTED** [review]

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot (hunk 4 above)
Comment (ChughShilpa) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot:29]: Added

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot (hunk 5 above)
Comment (ChughShilpa) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot:41]: Added

**Review by jgarciao — APPROVED** [review]

**Review by jiripetrlik — APPROVED** [review]

**Comment by ChughShilpa** [comment]
@jgarciao Could you please merge it ?

**Pull request merged**